### PR TITLE
Allow for going directly to an array page

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- This is preparatory work for a new generic list-and-loop pattern
- Add new page property "allowPathWithNoItems" so that we can go directly to a '/0' page in an array of items. The historical RJSF pattern required you to populate some of the array data first before you could navigate to the item's page.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#746

## Testing done

- Test for no regressions in existing forms
- browser, unit, integration tests

## Screenshots

n/a

## What areas of the site does it impact?

Should not affect any existing forms. New forms can use "allowPathWithNoItems".

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by powellkerry — APPROVED** [review]

**Review by Mottie — APPROVED** [review]
Nice! This looks like it'd fix the problem of needing the array pre-populated before allowing the use of the `showPagePerItem` setting. I like it! Thanks!

I haven't tested, but I wonder if this will fix the [problem with consecutive array pages](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/disability-benefits/all-claims/config/form.js#L362-L375).

File: src/platform/forms-system/src/js/helpers.js [hunk 1]
```diff
@@ -528,7 +528,15 @@ export function createUSAStateLabels(states) {
  * for each item in an array
  */
 function generateArrayPages(arrayPages, data) {
-  const items = get(arrayPages[0].arrayPath, data) || [];
+  let items = get(arrayPages[0].arrayPath, data) || [];
+
+  if (!items.length && arrayPages[0].allowPathWithNoItems) {
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/platform/forms-system/src/js/helpers.js:533]: nvm

**Review by tlei123 — APPROVED** [review]
LGTM

File: src/platform/forms-system/src/js/containers/FormPage.jsx [hunk 2]
```diff
@@ -53,6 +53,14 @@ class FormPage extends React.Component {
     const { pageConfig } = this.props.route;
     let newData = formData;
     if (pageConfig.showPagePerItem) {
+      if (
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/platform/forms-system/src/js/containers/FormPage.jsx:56]: unit-test found now.

**Pull request merged**